core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -471,17 +489,23 @@ private CatalogWrapper createCatalogWrapper(CatalogEntity entity) {
     Map<String, String> conf = entity.getProperties();
     String provider = entity.getProvider();
 
-    IsolatedClassLoader classLoader;
-    if (config.get(Configs.CATALOG_LOAD_ISOLATED)) {
-      String pkgPath = buildPkgPath(conf, provider);
-      String confPath = buildConfPath(conf, provider);
-      classLoader = IsolatedClassLoader.buildClassLoader(Lists.newArrayList(pkgPath, confPath));
-    } else {
-      // This will use the current class loader, it is mainly used for test.
-      classLoader =
-          new IsolatedClassLoader(
-              Collections.emptyList(), Collections.emptyList(), Collections.emptyList());
-    }
+    IsolatedClassLoader classLoader =
+        CLASS_LOADER_MAP.computeIfAbsent(
```

Comment:
on processing concurrent requests with same IsolatedClassloaderIdentifier , does it create only one classloader?

Reply:
Yes, the concurrent map can ensure it.